---
title: "Figure 1 Data Summary"
author: "Sara Gosline"
date: "10/23/2021"
output: html_document
---

## Get Data
This calls the standard data loading function that exists at `../beatAMLdata.R`. Once it is loaded it should not have to be reloaded. 

```{r setup, include=FALSE,message=FALSE}
if(!require(amlresistancenetworks)){
require(devtools)
devtools::install_github('sgosline/amlresistancenetworks')
require(amlresistancenetworks)
}

knitr::opts_chunk$set(echo = FALSE, warning=FALSE)


if(!exists('dataLoaded')){
  source("../beatAMLdata.R")
  loadBeatAMLData()
    dataLoaded=TRUE
    
  ##we're only working with 38 patients with drug data, so subset those
  pat.phos <- subset(pat.phos,Sample%in%auc.dat$`AML sample`)
  pat.data <- subset(pat.data,`AML sample`%in%auc.dat$`AML sample`)
    
  combined<-pat.phos%>%
    dplyr::rename(Phosphosite='LogFoldChange',`AML sample`='Sample')%>%
    right_join(pat.data)

  ##update phosphoSite categories
  pat.phos<-pat.phos%>%
    mutate(phTyr=stringr::str_detect(pat.phos$site,"Y[0-9]*y|y[0-9]*Y"),
           phSer=stringr::str_detect(pat.phos$site,"S[0-9]*s|s[0-9]*S"),
           phThr=stringr::str_detect(pat.phos$site,"T[0-9]*t|t[0-9]*T"))%>%
      pivot_longer(cols=c(phTyr,phSer,phThr),names_to='siteType',
                    values_to = 'bool')%>%
    subset(bool)%>%
    dplyr::select(-bool)
  
  
  KSDB <- read.csv(system.file('PSP&NetworKIN_Kinase_Substrate_Dataset_July2016.csv',
                               package='amlresistancenetworks'),
                    stringsAsFactors = FALSE)
  
  
  kdat<-KSDB%>%group_by(GENE)%>%select(SUB_GENE,SUB_MOD_RSD)%>%
    rowwise()%>%
    mutate(subval=paste(SUB_GENE,SUB_MOD_RSD,sep='-'))%>%
    mutate(phTyr=stringr::str_detect(SUB_MOD_RSD,'^Y'),
           phSer=stringr::str_detect(SUB_MOD_RSD,'^S'),
           phThr=stringr::str_detect(SUB_MOD_RSD,'^T'))%>%
    pivot_longer(cols=c(phTyr,phSer,phThr),
                  names_to='siteType',values_to = 'bool')%>%
    subset(bool)%>%
    dplyr::select(-bool)
  
}

```
We have loaded all the data and can begin to visualize


## Data Summary

The goal fo this markdown is to summarize the BeatAML data that we have so far. Patient summary, summary of AUC value, and general proteomics/protein data. Here we have the data for

```{r patient cohort, message=FALSE}
  res<-auc.dat%>%select(-c('FLT3-ITD','FLT3-MUT','NPM1'))%>%
    plotAllAUCs(.,pat.data,drug.metric='AUC',drug.column='Condition')

  full.drg<-res%>%
      tibble::rownames_to_column("AML sample")%>%
      left_join(auc.dat)%>%
      select(`AML sample`,Drug='Condition',proteins,RNA,mutations)%>%
    distinct()%>%
    group_by(Drug)%>%
    summarize(`Patients with proteomics`=length(which(proteins==TRUE)),`Patients with mRNA`=length(which(RNA==TRUE)),`Patients with Mutations`=length(which(mutations==TRUE)))
  
  knitr::kable(full.drg)
  write.csv(full.drg,'table1.csv')
  write.csv(res,'supp_table1.csv')
  
```

Many patients do not have complete data. What is the coverage of each data modality for each patient? How many genes are we getting?

```{r data coverage}

#For each patient, plot number of transcripts, WES mutations, phosphosites, proteins. 
###get total counts
protTotals <- pat.data%>%
  subset(proteinLevels!=0)%>%
  select(Gene)%>%unique()%>%nrow()

transTotals<-pat.data%>%
  subset(mRNALevels!=0.0)%>%
  select(Gene)%>%unique()%>%nrow()

mutTotals<-pat.data%>%
  subset(binaryMutations!=0.0)%>%
  select(Gene)%>%unique()%>%nrow()

phosTotals<-pat.phos%>%
  subset(LogFoldChange!=0)%>%
  select(site)%>%
  distinct()%>%nrow()
##get per-patient counts

np <- pat.data%>%
    group_by(`AML sample`)%>%
    subset(proteinLevels!=0)%>%
    summarize(Proteins=n())

nt <- pat.data%>%
      group_by(`AML sample`)%>%subset(mRNALevels!=0)%>%
      summarize(Transcripts=n())

ng <- pat.data%>%
      group_by(`AML sample`)%>%
      subset(binaryMutations!=0)%>%
      summarize(Mutations=n())

nph <- pat.phos%>%
      group_by(`Sample`)%>%
      subset(LogFoldChange!=0)%>%
      summarize(Phosphosites=n())%>%
      dplyr::rename(`AML sample`='Sample')

coverage <- np%>%full_join(nt)%>%
  full_join(ng)%>%
  full_join(nph)%>%
  tidyr::pivot_longer(cols=c(2:5),names_to='data type',values_to='number of features')

library(ggplot2)
ggplot(coverage,aes(x=`data type`,col=`data type`,y=`number of features`))+
  geom_jitter()+
  scale_y_log10()+
  ggtitle("Number of features across data types")

ggsave('featureSpaceSummary.pdf')

DT::datatable(data.frame(DataType=c('Transcripts','Mutations','Proteins','Phosphosites'),
           numFeatures=c(transTotals,mutTotals,protTotals,phosTotals)))

```

The datasets are indeed of varying size, and also have varying number of patients. Since many of the patients are incomplete, we save a second version of the data that has only those 'complete' datasets.

```{r full patients}


red.auc<-auc.dat%>%subset(`AML sample`%in%full.pats$`AML sample`)
red.dat<-pat.data%>%subset(`AML sample`%in%full.pats$`AML sample`)
red.phos<-pat.phos%>%subset(Sample%in%full.pats$`AML sample`)

knitr::kable(res[full.pats$`AML sample`,])
```



## Drug Family summary

We can also collate information by drug family to increase power. 


```{r drug family, message=FALSE, warning=FALSE}

  res<-auc.dat.fam%>%select(-c('FLT3-ITD','FLT3-MUT','NPM1'))%>%
    plotAllAUCs(.,pat.data,drug.metric='AUC',drug.column='family')

  full.drg<-res%>%
      tibble::rownames_to_column("AML sample")%>%
      left_join(auc.dat.fam)%>%
      select(`AML sample`,Drug='family',proteins,RNA,mutations)%>%
    distinct()%>%
    group_by(Drug)%>%
    summarize(`Patients with proteomics`=length(which(proteins==TRUE)),`Patients with mRNA`=length(which(RNA==TRUE)),`Patients with Mutations`=length(which(mutations==TRUE)))
  
  knitr::kable(full.drg)
  write.csv(full.drg,'fam_table1.csv')
  write.csv(res,'fam_supp_table1.csv')

```


## Protein/mRNA/phosphosite correlation values
how well do these samples correlate across all samples?


### mRNA/protein


```{r, prot cors,warning=FALSE,message=FALSE}

library(dplyr)
library(ggplot2)
mp<-combined%>%
  dplyr::select('AML sample','proteinLevels','mRNALevels','Gene')%>%
  distinct()%>%
  subset(mRNALevels!=0.0)%>%
  group_by(Gene)%>%
  summarize(corval=cor(proteinLevels,mRNALevels,method='spearman',use='pairwise.complete.obs'))%>%
  subset(!is.na(corval))

DT::datatable(mp)

meancor=mean(mp$corval)

mpp<-combined%>%
  dplyr::select('AML sample','proteinLevels','mRNALevels','Gene','binaryMutations')%>%
  distinct()%>%
  subset(mRNALevels!=0)%>%
  subset(proteinLevels!=0)%>%
  mutate(mRNALevels=log10(0.01+mRNALevels))%>%
  ggplot(aes(x=mRNALevels,y=proteinLevels))+geom_hex(bins=100)+scale_fill_viridis_c()+ggtitle(paste0('mRNA/Protein correlation: ',meancor))

ggsave('protMrnaCor.pdf',mpp)


```


No real correlation here

### protein/phosphosite

```{r prot phospho cor, warning=FALSE}
library(dplyr)
mp<-combined%>%
  dplyr::select('AML sample','proteinLevels','Phosphosite','Gene','site')%>%
  distinct()%>%
  group_by(site)%>%
  summarize(corval=cor(proteinLevels,Phosphosite,method='spearman',use='pairwise.complete.obs'))%>%
  subset(!is.na(corval))

DT::datatable(mp)
meancor=mean(mp$corval)

mpp<-combined%>%
  dplyr::select('AML sample','proteinLevels','Phosphosite','Gene','binaryMutations')%>%
  distinct()%>%
 # subset(mRNALevels>1)%>%
  subset(proteinLevels!=0)%>%
#  mutate(mRNALevels=log10(0.01+mRNALevels))%>%
  ggplot(aes(x=Phosphosite,y=proteinLevels))+geom_hex(bins=100)+scale_fill_viridis_c()+ggtitle(paste0('Protein/phosphosite correlation: ',meancor))

ggsave('protPhosho.pdf',mpp)


```

Now we can evaluate specific pathways.

## FLT3 pathway visualization

Let's see how each data type is represented across the FLT3 pathway

```{r flt3pathway, message=F,warning=F}
library(ggplot2)
flt3prots<-c("FLT3","FYN","GAB2","GRB2","HRAS",'HCK','KRAS','NRAS','PIK3R1','PTPN11','SOS1')


quiz.auc<-auc.dat%>%
#  subset(`AML sample`%in%flt3.pats$`AML sample`)%>%
  dplyr::select(Condition,AUC,'AML sample')%>%
  subset(Condition=='Quizartinib (AC220)')

flt3.pats<-combined%>%
  subset(Gene%in%flt3prots)%>%
  inner_join(quiz.auc)
  #subset(`AML sample`%in%auc.dat$`AML sample`)

genecors<-flt3.pats%>%
  group_by(Gene)%>%
  summarize(tcor=cor(mRNALevels,proteinLevels,use='pairwise.complete.obs',method='spearman'))

phoscors<-flt3.pats%>%
  
  group_by(Gene)%>%
  summarize(pcor=cor(Phosphosite,proteinLevels,use='pairwise.complete.obs',method='spearman'))

genecors<-genecors%>%left_join(phoscors)
write.csv(genecors,file='flt3CorVals.csv')

DT::datatable(genecors)

cor.genes<-genecors$Gene#$subset(genecors,abs(tcor)>0.25)$Gene

dotplots<-lapply(cor.genes,function(x)
  subset(flt3.pats,Gene==x)%>%
    mutate(mRNALevels=log10(0.01+mRNALevels))%>%
    mutate(mutated=as.factor(binaryMutations))%>%
    ggplot(aes(x=proteinLevels,y=mRNALevels,shape=mutated,color=AUC))+geom_point()+ggtitle(x)
)

p<-cowplot::plot_grid(plotlist=dotplots, ncol=2)
p

ggsave('flt3_correlatedDotPlots.pdf',p,width=12,height=24)

annotes<-quiz.auc%>%
  pivot_wider(values_from='AUC',names_from='Condition',values_fn=list(AUC=mean))%>%
  tibble::column_to_rownames('AML sample')

pat.order<-arrange(annotes, `Quizartinib (AC220)`)%>%rownames()
##get prot
prot.mat<-flt3.pats%>%
  dplyr::select(Gene,`AML sample`,proteinLevels)%>%
  pivot_wider(values_from = proteinLevels,names_from=`AML sample`,values_fill=0.0,
              values_fn=list(proteinLevels=mean))%>%
  mutate(Gene=paste(Gene,'prot'))%>%
  tibble::column_to_rownames('Gene')

##get phospho
ph.mat<-pat.phos%>%
    subset(Sample%in%auc.dat$`AML sample`)%>%
  subset(Gene%in%flt3prots)%>%
  dplyr::select(site,Sample,LogFoldChange)%>%
  pivot_wider(values_from='LogFoldChange',names_from='Sample',values_fill=0.0,
              values_fn=list(LogFoldChange=mean))%>%
  tibble::column_to_rownames('site')

## get mrna
t.mat<-flt3.pats%>%
  dplyr::select(Gene,`AML sample`,mRNALevels)%>%
  pivot_wider(values_from = mRNALevels,names_from=`AML sample`,values_fill=0.0,
              values_fn=list(mRNALevels=mean))%>%
  mutate(Gene=paste(Gene,'mRNA'))%>%
  tibble::column_to_rownames('Gene')
t.mat<-log10(0.01+t.mat)
missing=which(apply(t.mat,2,var)==0)

t.mat<-t.mat[,-missing]

##get VAF
v.mat<-combined%>%
  subset(Gene%in%c("FLT3","NRAS",'KRAS'))%>%
  dplyr::select(Gene,`AML sample`,binaryMutations)%>%
  pivot_wider(values_from = binaryMutations,names_from=`AML sample`,values_fill=0.0,
              values_fn=list(binaryMutations=max))%>%
  mutate(Gene=paste(Gene,'Gene'))%>%
  tibble::column_to_rownames('Gene')


flt3.vars<-t(v.mat) ##asave this for later so twe can add as ribbon to next figure
pat.order<-intersect(pat.order,colnames(t.mat))
fullmat<-rbind(prot.mat[,pat.order],t.mat[,pat.order],ph.mat[,pat.order])

fannotes<-cbind(Quizartinib=annotes[colnames(fullmat),],t(v.mat)[colnames(fullmat),])
#allz<-which(apply(fannotes,2,var)==0)
#fannotes<-fannotes[,-allz]

pheatmap(fullmat,cluster_rows = FALSE)

pheatmap(fullmat[sort(rownames(fullmat)),],
         cluster_rows = FALSE,cellheight=12,
         clustering_distance_cols = 'correlation',
         cluster_cols = TRUE, file='combined_heatmap.pdf',
         annotation_col = as.data.frame(fannotes) ,height=15)

pheatmap(as.matrix(prot.mat[,rownames(fannotes)]),
         cluster_rows = TRUE,cellheight=12,
        # clustering_distance_cols = 'correlation',
         cluster_cols = FALSE, file='prot_heatmap.pdf',
         annotation_col = as.data.frame(fannotes))
pheatmap(ph.mat[,rownames(fannotes)],
         cluster_rows = TRUE,cellheight=12,
         #clustering_distance_cols = 'correlation',
         cluster_cols = FALSE, file='phos_heatmap.pdf',
         annotation_col = as.data.frame(fannotes))
pheatmap(t.mat[,rownames(fannotes)],
         cluster_rows = TRUE,cellheight=12,
         #clustering_distance_cols = 'correlation',
         cluster_cols = FALSE, file='rna_heatmap.pdf',
         annotation_col = as.data.frame(fannotes))


dotplots<-lapply(subset(genecors,!is.na(pcor))$Gene,function(x)
  subset(flt3.pats,Gene==x)%>%
 #   mutate(mRNALevels=log10(0.01+mRNALevels))%>%
    mutate(mutated=as.factor(binaryMutations))%>%
    ggplot(aes(x=proteinLevels,y=Phosphosite,size=AUC,color=site))+geom_point()+ggtitle(x)
)

p<-cowplot::plot_grid(plotlist=dotplots, ncol=2)
p
ggsave('flt3_phosphoDotPlots.pdf',p,width=12,height=12)

```

This shows that proteomic data is important to understanding the impact of targeted therapies. 


## ERK pathway visualization

Let's see how each data type is represented across the ERK1/2 pathway

```{r erkpathway, message=F,warning=F}
library(ggplot2)
erk1prots<-c("CDK1","MAP2K1","IL6","IL6R","IL6ST",'JAK1','JAK2','MAPK3','TYK2','PTPN11')


quiz.auc<-auc.dat%>%
#  subset(`AML sample`%in%erk1.pats$`AML sample`)%>%
  dplyr::select(Condition,AUC,'AML sample')%>%
  subset(Condition=='Trametinib (GSK1120212)')

erk1.pats<-combined%>%
  subset(Gene%in%erk1prots)%>%
  inner_join(quiz.auc)
  #subset(`AML sample`%in%auc.dat$`AML sample`)

genecors<-erk1.pats%>%
  group_by(Gene)%>%
  summarize(tcor=cor(mRNALevels,proteinLevels,use='pairwise.complete.obs',method='spearman'))


phoscors<-erk1.pats%>%
  group_by(Gene)%>%
  summarize(pcor=cor(Phosphosite,proteinLevels,use='pairwise.complete.obs',method='spearman'))

genecors<-genecors%>%left_join(phoscors)

DT::datatable(genecors)
write.csv(genecors,file='erk1CorVals.csv')

cor.genes<-genecors$Gene#subset(genecors,abs(tcor)>0.1)$Gene

dotplots<-lapply(cor.genes,function(x)
  subset(erk1.pats,Gene==x)%>%
    mutate(mRNALevels=log10(0.01+mRNALevels))%>%
    mutate(mutated=as.factor(binaryMutations))%>%
    ggplot(aes(x=proteinLevels,y=mRNALevels,shape=mutated,color=AUC))+geom_point()+ggtitle(x)
)

p<-cowplot::plot_grid(plotlist=dotplots)
p

ggsave('erk1_correlatedDotPlots.pdf',p,width=12,height=12)

annotes<-quiz.auc%>%
  pivot_wider(values_from='AUC',names_from='Condition',values_fn=list(AUC=mean))%>%
  tibble::column_to_rownames('AML sample')

pat.order<-arrange(annotes, `Trametinib (GSK1120212)`)%>%rownames()
##get prot
prot.mat<-erk1.pats%>%
  dplyr::select(Gene,`AML sample`,proteinLevels)%>%
  pivot_wider(values_from = proteinLevels,names_from=`AML sample`,values_fill=0.0,
              values_fn=list(proteinLevels=mean))%>%
  mutate(Gene=paste(Gene,'prot'))%>%
  tibble::column_to_rownames('Gene')

##get phospho
ph.mat<-pat.phos%>%
    subset(Sample%in%auc.dat$`AML sample`)%>%
  subset(Gene%in%erk1prots)%>%
  dplyr::select(site,Sample,LogFoldChange)%>%
  pivot_wider(values_from='LogFoldChange',names_from='Sample',values_fill=0.0,
              values_fn=list(LogFoldChange=mean))%>%
  tibble::column_to_rownames('site')
## get mrna
t.mat<-erk1.pats%>%
  dplyr::select(Gene,`AML sample`,mRNALevels)%>%
  pivot_wider(values_from = mRNALevels,names_from=`AML sample`,values_fill=0.0,
              values_fn=list(mRNALevels=mean))%>%
  mutate(Gene=paste(Gene,'mRNA'))%>%
  tibble::column_to_rownames('Gene')
t.mat<-log10(0.01+t.mat)
missing=which(apply(t.mat,2,var)==0)

t.mat<-t.mat[,-missing]

##get VAF
v.mat<-erk1.pats%>%
  dplyr::select(Gene,`AML sample`,binaryMutations)%>%
  pivot_wider(values_from = binaryMutations,names_from=`AML sample`,values_fill=0.0,
              values_fn=list(binaryMutations=max))%>%
  mutate(Gene=paste(Gene,'Gene'))%>%
  tibble::column_to_rownames('Gene')


pat.order<-intersect(pat.order,colnames(t.mat))
fullmat<-rbind(prot.mat[,pat.order],t.mat[,pat.order],ph.mat[,pat.order])

fannotes<-data.frame(Trametinib=annotes[colnames(fullmat),1],flt3.vars[colnames(fullmat),])


#rownames(tannotes)<-colnames(fullmat)
#,t(v.mat)[colnames(fullmat),])
#allz<-which(apply(fannotes,2,var)==0)
#fannotes<-fannotes[,-allz]

pheatmap(fullmat,cluster_rows = FALSE)

pheatmap(fullmat[sort(rownames(fullmat)),],
         cluster_rows = FALSE,cellheight=12,
         clustering_distance_cols = 'correlation',
         cluster_cols = TRUE, file='erk1_combined_heatmap.pdf',
         annotation_col = as.data.frame(fannotes) ,height=15)

pheatmap(as.matrix(prot.mat[,rownames(fannotes)]),
         cluster_rows = TRUE,cellheight=12,
        # clustering_distance_cols = 'correlation',
         cluster_cols = FALSE, file='erk1_prot_heatmap.pdf',
         annotation_col = as.data.frame(fannotes))

pheatmap(ph.mat[,rownames(fannotes)],
         cluster_rows = TRUE,cellheight=12,
         #clustering_distance_cols = 'correlation',
         cluster_cols = FALSE, file='erk1_phos_heatmap.pdf',
         annotation_col = as.data.frame(fannotes))
pheatmap(t.mat[,rownames(fannotes)],
         cluster_rows = TRUE,cellheight=12,
         #clustering_distance_cols = 'correlation',
         cluster_cols = FALSE, file='erk1_rna_heatmap.pdf',
         annotation_col = as.data.frame(fannotes))


dotplots<-lapply(subset(genecors,!is.na(pcor))$Gene,function(x)
  subset(erk1.pats,Gene==x)%>%
 #   mutate(mRNALevels=log10(0.01+mRNALevels))%>%
    mutate(mutated=as.factor(binaryMutations))%>%
    ggplot(aes(x=proteinLevels,y=Phosphosite,size=AUC,color=site))+geom_point()+ggtitle(x)
)

p<-cowplot::plot_grid(plotlist=dotplots, ncol=2)
p
ggsave('erk1_phosphoDotPlots.pdf',p,width=12,height=12)

```


These figures suggest that mRNA and protein data are not highly correlated with drug response, as we'd expect, nor is phosphosite and protein data.


##Phosphosite coverage

We are curious to see how the various phosphosites are captured in the data

```{r phospho plots}

siteMap <- pat.phos%>%
  dplyr::select(site,siteType)%>%
  distinct()

sgenes<-tibble(site=unlist(siteMap$site))%>%
    tidyr::separate(site,
                    into=c('gene0','gene1','mod'),
                    sep='-',fill='left',remove=FALSE)%>% ##allow for dashed genes
    tidyr::unite(gene0,gene1,col='gene',sep='-')%>%
    mutate(gene=stringr::str_replace(gene,'^NA-',''))%>%
    tidyr::separate(mod,
                    into=c('mod1','mod2','mod3','mod4'),
                    extra='merge',sep='s|t|y')%>%
    tidyr::pivot_longer(cols=c(mod1,mod2,mod3,mod4),
                        names_to='var',values_to='mod')%>%
    dplyr::select(-var)%>%
    subset(mod!="")%>%
    subset(!is.na(mod))%>%
    tidyr::unite(gene,mod,col='singlesite',sep='-',remove=FALSE)%>%
    mutate(phTyr=stringr::str_detect(mod,'^Y'),
             phSer=stringr::str_detect(mod,'^S'),
             phThr=stringr::str_detect(mod,'^T'))%>%
    pivot_longer(cols=c(phTyr,phSer,phThr),
                  names_to='siteType',values_to = 'bool')%>%
    subset(bool)%>%
    dplyr::select(-bool)%>%
    unique()
  
siteCount<-sgenes%>%select(singlesite,siteType)%>%distinct()

print(paste(nrow(siteCount),'sites covered'))
##what is the coverage of site across amples?

sampCounts<-pat.phos%>%
  group_by(site)%>%
  summarize(nSamples=n_distinct(Sample))%>%
  left_join(sgenes)

library(ggplot2)

ggplot(sampCounts,aes(x=nSamples,fill=siteType))+geom_histogram()+scale_fill_viridis_d()+ggtitle('Phosphosites measured across samples')

ggsave('phosphositeCoverageAcrossPatients.pdf')

dtab<-siteCount%>%group_by(siteType)%>%
  summarize(totalSites=n_distinct(singlesite))
DT::datatable(dtab)

ggplot(siteCount,aes(x=siteType,fill=siteType))+
  geom_histogram(stat='count')+
  scale_fill_viridis_d()+
  ggtitle("Phosphosites measured by global phosphproteomics")

ggplot(kdat,aes(x=siteType,fill=siteType))+
  geom_histogram(stat='count')+
  scale_fill_viridis_d()+
  ggtitle('Phosphosites measured in PhophositeDB+NetworKIN')
```


Now that we have these stats we can compare how the different datasets compare.

```{r overlap}

combined <- sgenes%>%
  rename(subval='singlesite')%>%
  full_join(kdat)%>%
  dplyr::select(-GENE)%>%
  distinct()%>%
  rowwise()%>%
  mutate(annotation=ifelse(is.na(SUB_GENE),'NotInDB',ifelse(is.na(gene),'NotMeasured',"MeasuredAndannotated")))


ggplot(combined,aes(x=siteType,fill=annotation))+
  geom_histogram(stat='count',position='dodge')+
  scale_fill_viridis_d()+
  ggtitle('Phosphosites measured in data and PhophositeDB+NetworKIN')

ggsave("phosphositeCoverageInDBs.pdf")

```
