---
title: "Show cell line drug resistance"
author: "Sara Gosline"
date: "12/2/2020"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)


library(amlresistancenetworks)
library(wesanderson)
library(dplyr)
library(tidyr)
syn <- synapseLogin()

##assumes figure 2 was already generated!!!!
eval.preds<-readRDS(syn$get('syn26533741')$path)
all.preds<-readRDS(syn$get('syn26530501')$path)


full.preds<-all.preds%>%
  subset(compound%in%c("Quizartinib (AC220)","Trametinib (GSK1120212)"))%>%
  subset(Molecular%in%c('proteinLevels','Phosphosite','proteinLevels;Phosphosite'))%>%
  rename(var='compound')

#cl.preds<-rbind(cl.reg.preds,cl.lr.preds)

pal<-wes_palette('Darjeeling2')#,100,type='continuous')
```

## Cell Line resistance to drugs

As AML cells become resistant to drugs they undergo genetic, transcriptomic and proteomic changes that give rise to drug resistance. We have cultured two cell lines in the presence of Trametinib and Quizartinib and can evaluate the patient-derived signatures in these cell lines.


```{r collect cell line data, echo=FALSE,warning=FALSE}
tramProtData<-amlresistancenetworks::querySynapseTable('syn22986326')%>%
  subset(!is.nan(LogRatio))%>%
  mutate(Gene=unlist(Gene))%>%
  mutate(Molecular='proteinLevels')#%>%
#  subset(!sample%in%c('M13_PAR_01','M13_PAR_02','M13_PAR_03'))
 
tramPhosData<-amlresistancenetworks::querySynapseTable('syn24389738')%>%
  mutate(LogRatio=as.numeric(LogRatio))%>%
  subset(!is.nan(LogRatio))%>%
  mutate(Gene=unlist(Gene))%>%
  mutate(site=unlist(site))%>%
  mutate(Molecular='Phosphosite')#%>%
 # subset(!sample%in%c('M13_PAR_01','M13_PAR_02','M13_PAR_03'))

full.tram.prot<-tramPhosData%>%
  dplyr::select(Gene='site',LogRatio,sample,CellType,Molecular,Treatment)%>%
  rbind(dplyr::select(tramProtData,c(Gene,LogRatio,sample,CellType,Molecular,Treatment)))



quizProtData<-amlresistancenetworks::querySynapseTable('syn23595222')%>%
  subset(!is.nan(value))%>%
  mutate(Gene=unlist(Gene))%>%
  mutate(Molecular='proteinLevels')
 syn<-synapseLogin()
 
#quizPhosData<-syn$tableQuery('select * from syn24331593')$asDataFrame()   
quizPhosData<-amlresistancenetworks::querySynapseTable("syn24366514")%>%
  subset(!is.nan(value))%>%
  mutate(Gene=unlist(Gene))%>%
  mutate(site=unlist(site))%>%
  mutate(Molecular='Phosphosite')

full.quiz.prot<-quizPhosData%>%
  dplyr::select(Gene='site',LogRatio='value',sample='Sample',CellType='cellLine',
                Molecular,Treatment='Ligand')%>%
  rbind(dplyr::select(quizProtData,c(Gene,LogRatio='value',sample='Sample',
                                     CellType='cellLine',Molecular,Treatment='Ligand')))


```

Now we define functions to plot the results of the signatures on the cell line data

```{r plotting code}
#' spreadAndSelect
#' General plotting function
spreadAndSelect<-function(df,genes,mol, method,var){
  library(pheatmap)
  fname=paste0(method,'_preds_of_',var,'_with_',mol,'.pdf')

  ##update this
  termlist<-getFeaturesFromStringdf(genes,mol)
  
  res <-apply(termlist,1,function(x) {
    genes=unlist(strsplit(x[['genelists']], split=';')); 
    data.frame(Gene=genes,Datatype=rep(x[['DataType']],length(genes)))})%>%
    do.call(rbind,.)
  
  genelist=res$Gene
  
#  print(genelist)
  mol<-unique(res$Datatype)
  
  gene.d<-res%>%select(Gene,Datatype)%>%
      distinct()%>%
      tibble::column_to_rownames('Gene')
  
  #print(mol)
  
  if(length(genelist)<3)
    return(fname)

  sdf<-subset(df,Molecular%in%mol)

  mat<-sdf%>%
    select(Gene,LogRatio,sample)%>%
    replace_na(list(LogRatio=0.0))%>%
    subset(Gene%in%genelist)%>%
    tidyr::pivot_wider(values_from=LogRatio,
                       names_from=sample,
                  values_fn=list(LogRatio=sum),
                  values_fill=list(LogRatio=0.0))%>%
    tibble::column_to_rownames('Gene')
  
  #print(dim(mat))
  #print(mat)
  annotes<-sdf%>%
    select(CellType,sample)%>%distinct()%>%
    tibble::column_to_rownames('sample')
  
  dat.colors <- pal[1:4]
  names(dat.colors)<-c('Phosphosite','proteinLevels','mRNALevels','binaryMutations')
  
  annote.colors<-list(Datatype=dat.colors,CellType=wes_palette('Royal1',length(unique(sdf$CellType))))
  names(annote.colors$CellType)<-unique(sdf$CellType)
  #print(annotes)
  pheatmap(mat,annotation_col=annotes, cellheight=10,
           cellwidth=10,
           annotation_row=gene.d,
           filename=fname,clustering_method='ward.D2',
           annotation_colors = annote.colors, 
           clustering_distance_cols = 'euclidean', clustering_distance_rows='euclidean')#,color=pal)
  return(fname)
}


#'scatter plot
#' perhaps if we plot by scatter it will be better? 
scatterPlot<-function(df,genes,mol,method,var){
  library(ggplot2)
  fname=paste0(method,'_boxplot_preds_of_',var,'_with_',mol,'.pdf')
  print(fname)
  genelist<-getFeaturesFromString(genes,mol)
  sdf<-subset(df,Molecular==mol)%>%
    subset(Gene%in%genelist)
 # print(sdf)
  p1<-ggplot(sdf,aes(x=Gene,y=LogRatio,col=CellType))+
    geom_jitter()+geom_boxplot()+scale_color_manual(values=wes_palette('Darjeeling1',5))
  ggsave(fname,p1)
  
}
```

## Trametinib Analysis



```{r trametinib, echo=FALSE, warning=FALSE}
tram.preds<-subset(full.preds,var=="Trametinib (GSK1120212)")

pred.plots<-tram.preds%>%
  subset(numFeatures>2)%>%
  mutate(df=list(tibble::remove_rownames(full.tram.prot)))%>%
  dplyr::rename(mol='Molecular')%>%
  rowwise()%>%
  mutate(plotfile=spreadAndSelect(df,genes,mol,method,var))

red.tram.prot<-full.tram.prot%>%
  subset(Treatment%in%c('none','Trametinib Withdrawn'))#,'Trametinib'))

red.pred.plots<-tram.preds%>%subset(numFeatures>0)%>%
  mutate(df=list(tibble::remove_rownames(red.tram.prot)))%>%
  dplyr::rename(mol='Molecular')%>%
  mutate(method=paste0('red',method))%>%
  rowwise()%>%
  mutate(plotfile=spreadAndSelect(df,genes,mol,method,var))#%>%
 # mutate(scatter=scatterPlot(df,genes,mol,method,var))

```


## Quizartinib Analysis

```{r quizartinib, echo=FALSE, warning=FALSE}
quiz.preds<-subset(full.preds,var=="Quizartinib (AC220)")
pred.plots<-quiz.preds%>%subset(numFeatures>0)%>%
  mutate(df=list(tibble::remove_rownames(full.quiz.prot)))%>%
  rename(mol=Molecular)%>%
  rowwise()%>%
  mutate(plotfile=spreadAndSelect(df,genes,mol,method,var))

red.quiz.prot<-full.quiz.prot%>%
  subset(Treatment=='None')

red.pred.plots<-quiz.preds%>%subset(numFeatures>0)%>%
  mutate(df=list(tibble::remove_rownames(red.quiz.prot)))%>%
  rename(mol=Molecular)%>%
  mutate(method=paste0('red',method))%>%
  rowwise()%>%
  mutate(plotfile=spreadAndSelect(df,genes,mol,method,var))#%>%
#  mutate(scatter=scatterPlot(df,genes,mol,method,var))
```
Note that the `echo = FALSE` parameter was added to the code chunk to prevent printing of the R code that generated the plot.


### Networks
We compute networks from the combination of both proteomics and phosphoproteomics. this was done in figure 3.

