---
title: "Show cell line drug resistance"
author: "Sara Gosline"
date: "12/2/2020"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)


library(amlresistancenetworks)
library(wesanderson)
library(dplyr)
library(tidyr)

##assumes figure 2 was already generated!!!!
full.preds<-readRDS('combinedKfoldRes.rds')%>%
  subset(compound%in%c("Quizartinib (AC220)","Trametinib (GSK1120212)"))%>%
  subset(Molecular%in%c('proteinLevels','Phosphosite'))


#cl.preds<-rbind(cl.reg.preds,cl.lr.preds)

pal<-wes_palette('Darjeeling1',100,type='continuous')
```

## Cell Line resistance to drugs

As AML cells become resistant to drugs they undergo genetic, transcriptomic and proteomic changes that give rise to drug resistance. We have cultured two cell lines in the presence of Trametinib and Quizartinib and can evaluate the patient-derived signatures in these cell lines.



```{r collect cell line data, echo=FALSE,warning=FALSE}
tramProtData<-amlresistancenetworks::querySynapseTable('syn22986326')%>%
  subset(!is.nan(LogRatio))%>%
  mutate(Gene=unlist(Gene))%>%
  mutate(Molecular='proteinLevels')#%>%
#  subset(!sample%in%c('M13_PAR_01','M13_PAR_02','M13_PAR_03'))
 
tramPhosData<-amlresistancenetworks::querySynapseTable('syn24389738')%>%
  mutate(LogRatio=as.numeric(LogRatio))%>%
  subset(!is.nan(LogRatio))%>%
  mutate(Gene=unlist(Gene))%>%
  mutate(site=unlist(site))%>%
  mutate(Molecular='Phosphosite')#%>%
 # subset(!sample%in%c('M13_PAR_01','M13_PAR_02','M13_PAR_03'))

full.tram.prot<-tramPhosData%>%
  dplyr::select(Gene='site',LogRatio,sample,CellType,Molecular,Treatment)%>%
  rbind(dplyr::select(tramProtData,c(Gene,LogRatio,sample,CellType,Molecular,Treatment)))



quizProtData<-amlresistancenetworks::querySynapseTable('syn23595222')%>%
  subset(!is.nan(value))%>%
  mutate(Gene=unlist(Gene))%>%
  mutate(Molecular='proteinLevels')
 syn<-synapseLogin()
 
#quizPhosData<-syn$tableQuery('select * from syn24331593')$asDataFrame()   
quizPhosData<-amlresistancenetworks::querySynapseTable("syn24366514")%>%
  subset(!is.nan(value))%>%
  mutate(Gene=unlist(Gene))%>%
  mutate(site=unlist(site))%>%
  mutate(Molecular='Phosphosite')

full.quiz.prot<-quizPhosData%>%
  dplyr::select(Gene='site',LogRatio='value',sample='Sample',CellType='cellLine',
                Molecular,Treatment='Ligand')%>%
  rbind(dplyr::select(quizProtData,c(Gene,LogRatio='value',sample='Sample',
                                     CellType='cellLine',Molecular,Treatment='Ligand')))


```

```{r plotting code}
#' spreadAndSelect
#' General plotting function
spreadAndSelect<-function(df,genes,mol, method,var){
  library(pheatmap)
  fname=paste0(method,'_preds_of_',var,'_with_',mol,'.pdf')
<<<<<<< Updated upstream:pilotManuscript/fig67_cellLineResistance.Rmd
  print(fname)
  genelist<-getFeaturesFromString(genes,mol)
  #print(genelist)
=======
 # print(fname)
  
  ##update this
  genelist<-getFeaturesFromString(genes,mol)%>%unlist()
  if(length(genelist)<3)
    return(fname)
#print(genelist)
>>>>>>> Stashed changes:pilotManuscript/fig56_cellLineResistance.Rmd
  sdf<-subset(df,Molecular==mol)
  #print(sdf)
  mat<-sdf%>%
    select(Gene,LogRatio,sample)%>%
    replace_na(list(LogRatio=0.0))%>%
    subset(Gene%in%genelist)%>%
    tidyr::pivot_wider(values_from=LogRatio,names_from=sample,
                        values_fn=list(LogRatio=sum),values_fill=list(LogRatio=0.0))%>%
    tibble::column_to_rownames('Gene')
  #print(dim(mat))
  annotes<-sdf%>%
    select(CellType,sample)%>%distinct()%>%
    tibble::column_to_rownames('sample')
  annote.colors<-list(CellType=wes_palette('Royal1',length(unique(sdf$CellType))))
  names(annote.colors$CellType)<-unique(sdf$CellType)
  #print(annotes)
  pheatmap(mat,annotation_col=annotes, cellheight=10,cellwidth=10,
           filename=fname,clustering_method='ward.D2',annotation_colors = annote.colors, 
           clustering_distance_cols = 'euclidean', clustering_distance_rows='euclidean')#,color=pal)
  return(fname)
}


#'scatter plot
#' perhaps if we plot by scatter it will be better? 
scatterPlot<-function(df,genes,mol,method,var){
  library(ggplot2)
  fname=paste0(method,'_boxplot_preds_of_',var,'_with_',mol,'.pdf')
  print(fname)
  genelist<-getFeaturesFromString(genes,mol)
  sdf<-subset(df,Molecular==mol)%>%
    subset(Gene%in%genelist)
  print(sdf)
  p1<-ggplot(sdf,aes(x=Gene,y=LogRatio,col=CellType))+geom_jitter()+geom_boxplot()+scale_color_manual(values=wes_palette('Darjeeling1',5))
  ggsave(fname,p1)
  
}
```

## Trametinib Analysis



```{r trametinib, echo=FALSE, warning=FALSE}
tram.preds<-subset(full.preds,var=="Trametinib (GSK1120212)")

pred.plots<-tram.preds%>%subset(numFeatures>0)%>%
  mutate(df=list(tibble::remove_rownames(full.tram.prot)))%>%
  dplyr::rename(mol='Molecular')%>%
  rowwise()%>%
  mutate(plotfile=spreadAndSelect(df,genes,mol,method,var))

red.tram.prot<-full.tram.prot%>%
  subset(Treatment%in%c('none','Trametinib Withdrawn'))#,'Trametinib'))

red.pred.plots<-tram.preds%>%subset(numFeatures>0)%>%
  mutate(df=list(tibble::remove_rownames(red.tram.prot)))%>%
  dplyr::rename(mol='Molecular')%>%
  mutate(method=paste0('red',method))%>%
  rowwise()%>%
  mutate(plotfile=spreadAndSelect(df,genes,mol,method,var))#%>%
 # mutate(scatter=scatterPlot(df,genes,mol,method,var))

```


## Quizartinib Analysis

```{r quizartinib, echo=FALSE, warning=FALSE}
quiz.preds<-subset(full.preds,var=="Quizartinib (AC220)")
pred.plots<-quiz.preds%>%subset(numFeatures>0)%>%
  mutate(df=list(tibble::remove_rownames(full.quiz.prot)))%>%
  rename(mol=Molecular)%>%
  rowwise()%>%
  mutate(plotfile=spreadAndSelect(df,genes,mol,method,var))

red.quiz.prot<-full.quiz.prot%>%
  subset(Treatment=='None')

red.pred.plots<-quiz.preds%>%subset(numFeatures>0)%>%
  mutate(df=list(tibble::remove_rownames(red.quiz.prot)))%>%
  rename(mol=Molecular)%>%
  mutate(method=paste0('red',method))%>%
  rowwise()%>%
  mutate(plotfile=spreadAndSelect(df,genes,mol,method,var))#%>%
#  mutate(scatter=scatterPlot(df,genes,mol,method,var))
```
Note that the `echo = FALSE` parameter was added to the code chunk to prevent printing of the R code that generated the plot.


### Networks
We compute networks from the combination of both proteomics and phosphoproteomics


####Trametinib
```{r tram ,echo=FALSE,warning=FALSE}
fname=paste0('LASSSO-trametinib-preds')
print(fname)    

tr<-tram.preds%>%
  subset(method=='LASSO')%>%
<<<<<<< Updated upstream:pilotManuscript/fig67_cellLineResistance.Rmd
  select(genes,Molecular)
ph.genelist<-getFeaturesFromString(subset(tr,Molecular=='Phosphosite')$genes,'Phosphosite')
pr.genelist<-getFeaturesFromString(subset(tr,Molecular=='proteinLevels')$genes,'proteinLevels')
=======
  dplyr::select(genes,Molecular)

ph.genelist<-getFeaturesFromString(subset(tr,Molecular=='Phosphosite')$genes,'Phosphosite')%>%unlist()
pr.genelist<-getFeaturesFromString(subset(tr,Molecular=='proteinLevels')$genes,'proteinLevels')%>%unlist()
>>>>>>> Stashed changes:pilotManuscript/fig56_cellLineResistance.Rmd
  

    ##now we need to fix the gene list, since it's not going to match    
phgenes<-data.frame(genes=ph.genelist)%>%
  tidyr::separate(genes, into=c('gene','mod'),sep='-')%>%
  mutate(modlist=strsplit(mod,split='s|t|y'))%>%
  apply(1,function(x) paste(x$gene,x$modlist,sep='-'))%>%
  unlist()%>%unique()

phos.vals<-rep(1.5,length(phgenes))  
names(phos.vals)<-phgenes
  
prot.vals<-rep(1,length(pr.genelist))
names(prot.vals)<-pr.genelist

#het=computePhosphoNetwork(phos.vals,prot.vals,nrand=100,beta=2,fname)


##now do the same for logistic
fname=paste0('Logistic-trametinib-preds')
print(fname)    

tr.preds<-tram.preds%>%
  subset(method=='LogisticReg')%>%
  select(genes,Molecular)

ph.genelist<-getFeaturesFromString(subset(tr.preds,Molecular=='Phosphosite')$genes,'Phosphosite')%>%unlist()
pr.genelist<-getFeaturesFromString(subset(tr.preds,Molecular=='proteinLevels')$genes,'proteinLevels')%>%unlist()
  

    ##now we need to fix the gene list, since it's not going to match    
phgenes<-data.frame(genes=ph.genelist)%>%
  separate(genes, into=c('gene','mod'),sep='-')%>%
  mutate(modlist=strsplit(mod,split='s|t|y'))%>%
  apply(1,function(x) paste(x$gene,x$modlist,sep='-'))%>%
  unlist()%>%unique()

phos.vals<-rep(1.5,length(phgenes))  
names(phos.vals)<-phgenes
  
prot.vals<-rep(1,length(pr.genelist))
names(prot.vals)<-pr.genelist

#het=computePhosphoNetwork(phos.vals,prot.vals,nrand=100,beta=2,fname)
#library(PCSF)
#plot.PCSF(het)

```
 
#### Quizartinib
```{r quiz networks,echo=FALSE,warning=FALSE}
fname=paste0('LASSSO-quizartinib-preds')
print(fname)    

qu.preds<-quiz.preds%>%
  subset(method=='LASSO')%>%
  select(genes,Molecular)

ph.genelist<-getFeaturesFromString(subset(qu.preds,Molecular=='Phosphosite')$genes,'Phosphosite')%>%unlist()
pr.genelist<-getFeaturesFromString(subset(qu.preds,Molecular=='proteinLevels')$genes,'proteinLevels')%>%unlist()

    ##now we need to fix the gene list, since it's not going to match    
phgenes<-data.frame(genes=ph.genelist)%>%
  separate(genes, into=c('gene','mod'),sep='-')%>%
  mutate(modlist=strsplit(mod,split='s|t|y'))%>%
  apply(1,function(x) paste(x$gene,x$modlist,sep='-'))%>%
  unlist()%>%unique()

phos.vals<-rep(1.5,length(phgenes))  
names(phos.vals)<-phgenes
  
prot.vals<-rep(1,length(pr.genelist))
names(prot.vals)<-pr.genelist

het=computePhosphoNetwork(phos.vals,prot.vals,nrand=100,beta=4,fname)
#library(PCSF)
#plot.PCSF(het)

#logistic preds
fname=paste0('Logistic-quizartinib-preds')
print(fname)    

qu.preds<-quiz.preds%>%
  subset(method=='LogisticReg')%>%
  select(genes,Molecular)
ph.genelist<-getFeaturesFromString(subset(qu.preds,Molecular=='Phosphosite')$genes,'Phosphosite')%>%unlist()
pr.genelist<-getFeaturesFromString(subset(qu.preds,Molecular=='proteinLevels')$genes,'proteinLevels')%>%unlist()
  

    ##now we need to fix the gene list, since it's not going to match    
phgenes<-data.frame(genes=ph.genelist)%>%
  separate(genes, into=c('gene','mod'),sep='-')%>%
  mutate(modlist=strsplit(mod,split='s|t|y'))%>%
  apply(1,function(x) paste(x$gene,x$modlist,sep='-'))%>%
  unlist()%>%unique()

phos.vals<-rep(1.5,length(phgenes))  
names(phos.vals)<-phgenes
  
prot.vals<-rep(1,length(pr.genelist))
names(prot.vals)<-pr.genelist

#het=computePhosphoNetwork(phos.vals,prot.vals,nrand=100,beta=2,fname)
```

That's the end!
